lib/whitehall_importer.rb

```diff
@@ -23,9 +25,26 @@ def self.create_migration(organisation_content_id, document_type)
     end
   end
 
-  def self.import_and_sync(whitehall_document)
-    whitehall_import = WhitehallMigration::DocumentImport.create!(
-      whitehall_document_id: whitehall_document["id"],
+  def self.queue_migration_jobs(whitehall_migration)
+    documents = WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id)
+    documents.each do |document|
+      WhitehallDocumentImportJob.perform_later(document)
+    end
+  end
+
+  def self.mark_migration_completed(whitehall_migration_document_import)
+    running_states = %w(pending importing imported syncing)
+    whitehall_migration = WhitehallMigration.find(whitehall_migration_document_import.whitehall_migration_id)
+    if WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id, state: running_states).empty?
+      whitehall_migration.update!(
+        end_time: Time.current,
+      )
+    end
+  end
+
+  def self.import_and_sync(whitehall_import)
+    whitehall_document = GdsApi.whitehall_export.document_export(whitehall_import.whitehall_document_id).to_h
```

Comment:
I can see why this has moved to here but one thing that becomes a bit unclear is who/what is responsible for dealing with whether this fails. This was a bit simpler when this was a rake task as we'd just have that task fail. Maybe this is something to be concerned about with retrying as I'm a little unsure of a clear answer - just feels a riskier introduction.

Reply:
We've left a comment in https://trello.com/c/xXNlisKk/1273-spike-how-and-when-should-we-retry-jobs to make sure that this is followed up when that spike gets done :)